pallets/subtensor/src/block_step.rs

```diff
@@ -82,6 +84,43 @@ impl<T: Config> Pallet<T> {
         LoadedEmission::<T>::get(netuid).unwrap()
     }
 
+    // Determines if due for a halving at the current block and
+    // updates block emissions if so.
+    //
+    pub fn check_halving() {
+        let total_issuance = Self::get_total_issuance();
+        let current_emission = Self::get_block_emission();
+        let last_halving_issuance = Self::get_last_halving_issuance();
+
+        let is_halving: bool = match total_issuance {
```

Comment:
Since the halving occurs every 10_500_000 TAOs , do you think we can avoid the matches and do this instead?

```
pub fn check_halving() {
    let total_issuance = Self::get_total_issuance();
    let current_emission = Self::get_block_emission();
    let last_halving_issuance = Self::get_last_halving_issuance();

    // Assuming halving occurs every 10,500,000 tokens.
    const HALVING_INTERVAL: u64 = 10_500_000;
    // Calculate the number of halvings that should have occurred by now.
    let expected_halvings = total_issuance / HALVING_INTERVAL;
    // Calculate the number of halvings that had occurred by the last recorded issuance.
    let actual_halvings = last_halving_issuance / HALVING_INTERVAL;

    // Check if a new halving event should occur.
    if expected_halvings > actual_halvings {
        if current_emission == 0 {
            // Halvings are already complete, or no emission to halve.
            return;
        }
        // Halve the current emission and update the last halving issuance.
        let new_emission = current_emission / 2;
        Self::set_block_emission(new_emission);
        // Update the last halving issuance to the current total issuance.
        Self::set_last_halving_issuance(total_issuance);
        log::info!(
            "Halving event at total issuance {}: New block emission is {}",
            total_issuance,
            new_emission
        );
    }
}
```

Reply:
yeah this approach is better